edit/edit_test.go

```diff
@@ -58,15 +58,21 @@ var shortenLabelTests = []struct {
 	result string
 }{
 	{"//devtools/buildozer:rule", "devtools/buildozer", ":rule"},
+	{"@//devtools/buildozer:rule", "devtools/buildozer", ":rule"},
 	{"//devtools/buildozer:rule", "devtools", "//devtools/buildozer:rule"},
 	{"//base:rule", "devtools", "//base:rule"},
 	{"//base:base", "devtools", "//base"},
 	{"//base", "base", ":base"},
+	{"//devtools/buildozer:buildozer", "", "//devtools/buildozer"},
 	{"@r//devtools/buildozer:buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools/buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools", "devtools", "@r//devtools"},
 	{"@r:rule", "", "@r:rule"},
 	{"@r", "", "@r"},
+	{"@foo//:foo", "", "@foo"},
+	{"@foo//devtools:foo", "", "@foo//devtools:foo"},
```

Comment:
this looks out of place, same LHS as the next case, but empty 'pkg' here looks wrong.

Reply:
As far as I know, we are referring to external repo while we are in the current workspace root. For the next case, we are also referring to external repo but we want to make sure even though we have the same package name, we shouldn't shorten it anyway.

Could you elaborate more what looks wrong to you? Thanks.

Reply:
only one of
{"@foo//devtools:foo", "", "@foo//devtools:foo"},
{"@foo//devtools:foo", "devtools", "@foo//devtools:foo"},

should work, as with the same input you expect the same output -- these are 2 tests with the same input but 2 different outputs.